packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/shared/base-load-balancer.ts

```diff
@@ -240,28 +315,61 @@ export abstract class BaseLoadBalancer extends Resource {
 
     const internetFacing = ifUndefined(baseProps.internetFacing, false);
 
-    const vpcSubnets = ifUndefined(baseProps.vpcSubnets,
-      (internetFacing ? { subnetType: ec2.SubnetType.PUBLIC } : {}) );
-    const { subnetIds, internetConnectivityEstablished } = baseProps.vpc.selectSubnets(vpcSubnets);
+    if (baseProps.vpcSubnets && additionalProps.subnetMappings) {
+      throw new ValidationError('`vpcSubnets` and `subnetMappings` cannot be specified at the same time.', this);
+    }
 
-    this.vpc = baseProps.vpc;
+    let subnetIds: string[] | undefined;
+    let subnetMappings: SubnetMapping[] | undefined = additionalProps.subnetMappings;
+    let internetConnectivityEstablished: IDependable | undefined;
 
     if (additionalProps.ipAddressType === IpAddressType.DUAL_STACK_WITHOUT_PUBLIC_IPV4 &&
-      additionalProps.type !== cxschema.LoadBalancerType.APPLICATION) {
+        additionalProps.type !== cxschema.LoadBalancerType.APPLICATION) {
       throw new ValidationError(`'ipAddressType' DUAL_STACK_WITHOUT_PUBLIC_IPV4 can only be used with Application Load Balancer, got ${additionalProps.type}`, this);
     }
 
+    this.vpc = baseProps.vpc;
+
+    if (subnetMappings) {
+      if (internetFacing) {
+        const mappedSubnetIds = subnetMappings.map(mapping => mapping.subnet.subnetId);
+        internetConnectivityEstablished = baseProps.vpc.selectSubnets(
+          { subnetFilters: [ec2.SubnetFilter.byIds(mappedSubnetIds)] },
+        ).internetConnectivityEstablished;
```

Comment:
nit: could we rename this variable so it has a different name from the actual property that we're checking when filtering. It feels a bit confusing if they're named the same.

```suggestion
        internetConnectivityEstablishedSubnets = baseProps.vpc.selectSubnets(
          { subnetFilters: [ec2.SubnetFilter.byIds(mappedSubnetIds)] },
        ).internetConnectivityEstablished;
```